Sources/IRGen/Expression/IRBinaryExpression.swift

```diff
@@ -30,26 +30,35 @@ struct IRBinaryExpression {
     let rhs = IRExpression(expression: binaryExpression.rhs, asLValue: asLValue)
       .rendered(functionContext: functionContext)
 
+    var preamble = lhs.preamble + "\n" + rhs.preamble
+    let lhsExp = lhs.expression
+    let rhsExp = rhs.expression
+    var code = ""
```

Comment:
You assign to this in every branch of the switch, so you should make this a `let` (constant) with no default value.

```
let code: String
switch binaryExpression.opToken {
// bla
```